libethereum/Executive.cpp

```diff
@@ -373,6 +391,11 @@ bool Executive::go(OnOpFunc const& _onOp)
 			auto vm = _onOp ? VMFactory::create(VMKind::Interpreter) : VMFactory::create();
 			if (m_isCreation)
 			{
+				if (m_envInfo.number() >= m_sealEngine.chainParams().u256Param("metropolisForkBlock") &&
+					m_s.addressHasCode(m_newAddress) &&
+					(m_s.code(m_newAddress).size() != 0 || m_s.getNonce(m_newAddress)))
```

Comment:
This doesn't look right. We need to fail if either code is non-empty or nonce is non-zero, right?
Then it should be `(m_s.addressHasCode(m_newAddress) || m_s.getNonce(m_newAddress) != 0)`